src/objective/multiclass_obj.cu

```diff
@@ -59,14 +59,17 @@ class SoftmaxMultiClassObj : public ObjFunction {
     const int nclass = param_.num_class;
     const auto ndata = static_cast<int64_t>(preds.Size() / nclass);
 
-    auto devices = GPUSet::All(tparam_->gpu_id, tparam_->n_gpus, preds.Size());
-    out_gpair->Shard(GPUDistribution::Granular(devices, nclass));
-    info.labels_.Shard(GPUDistribution::Block(devices));
-    info.weights_.Shard(GPUDistribution::Block(devices));
-    preds.Shard(GPUDistribution::Granular(devices, nclass));
+    GPUSet devices = (preds.Devices().IsEmpty() && tparam_->external_memory)
```

Comment:
What's wrong with resharding to empty device?

Reply:
possibly nothing. however, sharding without any devices (or to no device) doesn't seem right, and hence i made this change.